qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500_module.py

```diff
@@ -243,6 +243,24 @@ def get_measurement_summary(status_array: np.ndarray) -> str:
 
     return summary
 
+
+def convert_dummy_val_to_nan(param: _FMTResponse):
+    """
+    Converts dummy value to NaN. Instrument may output dummy value (
+    199.999E+99) if measurement data is over the measurement range. Or the
+    sweep measurement was aborted by the automatic stop function or power
+    compliance. Or if any abort condition is detected. Dummy data
+    199.999E+99 will be returned for the data after abort."
+
+    Args:
+        param: This must be of type named tuple _FMTResponse.
+
+    """
+    for index, value in enumerate(param.value):
```

Comment:
not sure if we care about performance, or if my suggestion would even help, but we might be able to save some time by converting to numpy array, using logical slicing there, and then covnerting back to a list:
```python
values = np.array(param.value)
values[values > 199.999e99] = float('nan')
param.value = values.tolist()
```

Reply:
Since these are namedtuple I cannot assign values to its attribute value. However if this is faster I can define a new namedtuple replacing the value of 'value'. 
> param = param._replace(value = values.tolist()) 

Does this solution sound ok to you?

Reply:
nah, let's not use private methods :) not worth it